homeworkW11D02/src/ba/bitcamp/homeworkW11D02/Server.java

```diff
@@ -0,0 +1,234 @@
+package ba.bitcamp.homeworkW11D02;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Desktop;
+import java.awt.Font;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import javax.swing.BorderFactory;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+
+/**
+ * @author Gordan
+ *
+ *         This class represents chat server. Server contains one JFrame window
+ *         with tree panels on it. Upper panel contains JTextArea where chat
+ *         between client and server is shown, lower panel contains JTextField
+ *         where server user writes his messages and main panel contains upper
+ *         and lower panels. There is also one JScrollPane on the text area.
+ */
+public class Server extends JFrame {
+
+	private static final long serialVersionUID = 6866315603565348167L;
+
+	// Window components
+	private JPanel mainPanel = new JPanel();
+	private JPanel downPanel = new JPanel();
+	private JPanel upPanel = new JPanel();
+	private JTextArea textArea = new JTextArea();
+	private JTextField textField = new JTextField();
+	private JScrollPane scrollPane = new JScrollPane(textArea,
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+
+	// Server and regular socket
+	private ServerSocket server;
+	private Socket client;
+
+	// List of directories
+	private ArrayList<String> listOfDirectories;
+
+	/**
+	 * Constructor for creating Server chat
+	 */
+	public Server() {
+
+		/*
+		 * Adding KeyListener to text field
+		 */
+		textField.addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+
+				/*
+				 * If server user types ENTER key on the keyboard and if the
+				 * text field is not empty, written message will be sent to a
+				 * client
+				 */
+				if (e.getKeyCode() == KeyEvent.VK_ENTER
+						&& !textField.getText().equals("")) {
+					try {
+
+						BufferedWriter writer = new BufferedWriter(
+								new OutputStreamWriter(client.getOutputStream()));
+						writer.write("Server: " + textField.getText());
```

Comment:
Why send the 'Server: ' prefix in the message, client should know that messages coming are server messages.